### PR TITLE
CCD-1604 : Fortify Scan implementation for HMI Inbound Adapter

### PR DESCRIPTION
**Pull request by sumansreenivasan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-1604

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: Jenkinsfile_nightly [hunk 1]
```diff
@@ -11,7 +11,12 @@ def type = "java"
 def product = "rpe"
```
Comment (MSancaktutar) [inline Jenkinsfile_nightly:11]: @sumansreenivasan  - Could you please correct the product name as `hmc`?

File: Jenkinsfile_nightly [hunk 2]
```diff
@@ -11,7 +11,12 @@ def type = "java"
...
 def component = "spring-boot-template"
 
-withNightlyPipeline(type, product, component) {}
+withNightlyPipeline(type, product, component) {
+  enableFortifyScan('ccd-aat')
```
Comment (MSancaktutar) [inline Jenkinsfile_nightly:15]: we have `hmc-aat` vault. so, better to use that one.

File: config/fortify-client.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+fortify.client.releaseId=73284
```
Comment (MSancaktutar) [inline config/fortify-client.properties:1]: i assume this is grabbed from Fortify, already.

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
just 2 tiny bots to be actioned please.

File: config/fortify-client.properties (hunk 3 above)
Comment (sumansreenivasan) [reply config/fortify-client.properties:1]: Yes, Kaan provided me with this.

File: Jenkinsfile_nightly [hunk 4]
```diff
@@ -8,10 +8,15 @@ properties([
 @Library("Infrastructure")
 
 def type = "java"
-def product = "rpe"
+def product = "hmc"
 def component = "spring-boot-template"
 
-withNightlyPipeline(type, product, component) {}
+withNightlyPipeline(type, product, component) {
+  enableFortifyScan('hmc-aat')
```
Comment (kaanaktas) [inline Jenkinsfile_nightly:15]: You don't need to pass parameter, if it is the same with product name.

Comment (kaanaktas) [reply Jenkinsfile_nightly:15]: I am also seeing below error. This vault needs to be defined in Azure.

org.jenkinsci.plugins.azurekeyvaultplugin.AzureKeyVaultException: Failed to retrieve secret fortify-on-demand-username from vault https://hmc-aat.vault.azure.net, error message: Status code 404, "{"error":{"code":"SecretNotFound","message":"A secret with (name/id) fortify-on-demand-username was not found in this key vault. If you recently deleted this secret you may be able to recover it using the correct recovery command. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125182"}}"

**Review by MSancaktutar — APPROVED** [review]
lgtm

**Pull request merged**